cli/file_fetcher.rs

```diff
@@ -127,79 +123,81 @@ impl SourceFileFetcher {
 
     // If file is not in memory cache check if it can be found
     // in local cache - which effectively means trying to fetch
-    // using "--cached-only" flag. We can safely block on this
-    // future, because it doesn't do any asynchronous action
-    // it that path.
-    let fut = self.get_source_file_async(specifier.as_url(), true, false, true);
-
-    futures::executor::block_on(fut).ok()
+    // using "--cached-only" flag.
+    // It should be safe to for caller block on this
+    // future, because it doesn't actually do any asynchronous
+    // action in that path.
+    self
+      .get_source_file_async(specifier.as_url(), true, false, true)
+      .await
+      .ok()
```

Comment:
It seems like if you're fetching a cached source file, you should be using sync ops. I see that it was using block_on before, so this is better. However, is there not a way to load this synchronously?

Reply:
To be determined - most likely this will be completely removed, when I get to untangle TsCompiler from `file_fetcher`

Reply:
Could you add a TODO here that fetching cache should use blocking fs syscalls?

Reply:
Done